corehq/apps/saved_reports/tests/test_models.py

```diff
@@ -1,18 +1,65 @@
+from datetime import datetime, timedelta
+from django.http.request import QueryDict
 from django.conf import settings
 from django.test import SimpleTestCase, TestCase
 from smtplib import SMTPSenderRefused
+from dimagi.utils.dates import DateSpan
 from dimagi.utils.django.email import LARGE_FILE_SIZE_ERROR_CODE
 from unittest.mock import create_autospec, patch, PropertyMock, ANY
 from corehq.apps.reports import views
 from corehq.apps.users.models import CouchUser, WebUser
 from corehq.apps.saved_reports import models
+from corehq.apps.saved_reports.tasks import create_config_for_email
 from ..models import ReportNotification, ReportConfig
 
 
 class TestReportConfig(TestCase):
 
-    def tearDown(self) -> None:
-        self.config.delete()
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        domain = 'test-domain'
+        user_id = 'nothing'
+        report_slug = 'case_list_explorer'
+        report_type = 'project_report'
+
+        GET = {'send_to_owner': 'true',
+               'subject': 'Case List Explorer',
+               'getReportRenderUrl':
+                   '/a/test-domain/reports/undefined/case_list_explorer/?search_xpath=case_name%3D%22EXAMPLEXPATH'
+                   '%22&explorer_columns=%5B%7B%22name%22%3A%22%40case_type%22%2C%22label%22%3A%22%40case_type'
+                   '%22%7D%2C%7B%22name%22%3A%22case_name%22%2C%22label%22%3A%22case_name%22%7D%2C%7B%22name'
+                   '%22%3A%22last_modified%22%2C%22label%22%3A%22last_modified'
+                   '%22%7D%5D&case_list_filter=project_data&case_type=&is_open=',
+               'params':
+                   'search_xpath=case_name%3D%22EXAMPLEXPATH%22&explorer_columns=%5B%7B%22name'
+                   '%22%3A%22%40case_type%22%2C%22label%22%3A%22%40case_type%22%7D%2C%7B%22name'
+                   '%22%3A%22case_name%22%2C%22label%22%3A%22case_name%22%7D%2C%7B%22name'
+                   '%22%3A%22last_modified%22%2C%22label%22%3A%22last_modified'
+                   '%22%7D%5D&case_list_filter=project_data&case_type=&is_open='
+               }
+        GET_dict = QueryDict('', mutable=True)
+        GET_dict.update(GET)
+        test_date = datetime(2023, 7, 25)
+
+        request_data = {
+            'GET': GET_dict,
+            'META': {'QUERY_STRING': '', 'PATH_INFO': '/a/test-domain/reports/email_onceoff/case_list_explorer/'},
+            'datespan': DateSpan(
+                startdate=test_date - timedelta(days=30),
+                enddate=test_date
+            ),
+            'couch_user': user_id,
+            'can_access_all_locations': True
+        }
+
+        cls.config = create_config_for_email(report_type, report_slug, user_id, domain, request_data)
+        cls.config.save()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.config.delete()
+        super().tearDownClass()
```

Comment:
nit: safer and more succinct clean up
```suggestion
        cls.addClassCleanup(cls.config.delete)
```

`addClassCleanup` is normally safer than `tearDownClass` because it will be called even if `setUpClass` raises an exception (`tearDownClass` would not be called in that scenario). However, in this case it doesn't make much difference since it is the last statement in the method. However, it is more future proof, so still recommended.